userapi/src/main/java/fk/prof/userapi/model/Tree.java

```diff
@@ -0,0 +1,23 @@
+package fk.prof.userapi.model;
+
+/**
+ * Created by gaurav.ashok on 01/06/17.
+ */
+public interface Tree<T> {
+
+    T get(int idx);
+
+    int getChildrenSize(int idx);
```

Comment:
Is this necessary? Couldn't the user always do get(idx) and then call appropriate childCount method? Of course that assumes frame node has childcount getter but that seems a fair assumption. Usually any algo would have fetched the node already so a redundant hash map lookup again for getting childcount may not be necessary.

Still, your call on this one.

Reply:
Kept the method since the usages do not have the node fetched before calling the childrenSize method.